### PR TITLE
JENKINS-54840 Stop calling get log file

### PR DESCRIPTION
**Pull request by timja**
* Adds a new action into the sidebar "Failure Scan Log", 
* removes the BFA prefix to log messages as now in its own log file
* harcodes the log file name to be "log" is it always seems to be
image

### CODE REVIEW

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+/**
+ * The Action for adding a link to the analysis for each run.
+ */
+public class ScanLogAction implements Action {
```
Comment (mwinter69) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java]: Should implement RunAction2 so you still have the association after a Restart of Jenkins

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * Log file name.
+     */
+    public static final String FILE_NAME = "com.sonyericsson.jenkins.plugins.bfa.ScanLogAction.log";
+
+    private final Run run;
```
Comment (mwinter69) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java]: Should be transient

File: src/main/resources/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction/index.jelly [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout" xmlns:st="jelly:stapler">
+    <l:layout title="Failure Cause Scan Log">
+        <st:include page="sidepanel.jelly" it="${it.run}"/>
+        <l:main-panel>
+            <h1>
+                ${%Failure Cause Scan Log}
+            </h1>
+            <pre>${it.getLogText()}</pre>
```
Comment (jglick) [inline src/main/resources/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction/index.jelly]: Not the best UI but good enough to start!

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * Log text for the analysis.
+     * @return the log text, lines are separated by \n
+     * @throws IOException if the log can't be found
+     */
+    public String getLogText() throws IOException {
+        return Files.lines(new File(run.getRootDir(), FILE_NAME).toPath())
+                .collect(Collectors.joining("\n"));
```
Comment (jglick) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java]: I think you can just use

```java
return FileUtils.toString(new File(run.getRootDir(), FILE_NAME), StandardCharsets.UTF_8);
```

Comment (timja) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java]: I assume you meant `readFileToString` pushed that

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java [hunk 5]
```diff
@@ -74,7 +73,7 @@ public FoundIndication scan(Run build) throws IOException {
             List<FoundFailureCause> foundFailureCauses = FailureReader.scanSingleLinePatterns(causes,
                                                                                               build,
                                                                                               reader,
-                                                                                              currentFile);
+                                                                                              "log");
```
Comment (rsandell) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java]: I can't remember what that file parameter is used for, a quick check in the current code this seems to be harmless. Maybe it's used for statistics or something?

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java [hunk 6]
```diff
@@ -231,33 +242,33 @@ private static void printDownstream(PrintStream buildLog, List<FailureCauseDispl
...
 
-        buildLog.println("[BFA] Scanning build for known causes...");
+        buildLog.println("Scanning build for known causes...");
         long start = System.currentTimeMillis();
         final List<FoundFailureCause> foundFailureCauseList = findIndications(causes, build, buildLog);
 
         long time = System.currentTimeMillis() - start;
         if (logger.isLoggable(Level.FINER)) {
-            logger.log(Level.FINER, "[BFA] [{0}] {1}ms", new Object[]
+            logger.log(Level.FINER, "[{0}] {1}ms", new Object[]
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java]: exactly... can revert if you want, but it's unneeded

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java (hunk 5 above)
Comment (timja) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java]: not sure, it was always called "log" when I checked though.

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java [hunk 7]
```diff
@@ -104,7 +106,7 @@ public void onStarted(Run build, TaskListener listener) {
             } catch (IOException e) {
                 logger.log(Level.WARNING, "Failed to add a ScannerJobProperty to "
                         + build.getParent().getFullDisplayName(), e);
-                listener.getLogger().println("[BFA] WARNING! Failed to add the scanner property to this job.");
+                listener.getLogger().println("WARNING! Failed to add the scanner property to this job.");
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java]:
```suggestion
                listener.getLogger().println("[BFA] WARNING! Failed to add the scanner property to this job.");
```

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (rsandell) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java]: no need to revert

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java (hunk 5 above)
Comment (rsandell) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java]: maybe it should be a constant then?

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java [hunk 8]
```diff
@@ -231,33 +242,33 @@ private static void printDownstream(PrintStream buildLog, List<FailureCauseDispl
         threadPoolExecutor.setCorePoolSize(PluginImpl.getInstance().getNrOfScanThreads());
         threadPoolExecutor.setMaximumPoolSize(PluginImpl.getInstance().getNrOfScanThreads());
 
-        buildLog.println("[BFA] Scanning build for known causes...");
+        buildLog.println("Scanning build for known causes...");
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java]: done both

screenshot:
image

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java (hunk 5 above)
Comment (timja) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/BuildLogFailureReader.java]: done

**Comment by timja** [comment]
All comments resolved please re-review

**Comment by jonsten** [comment]
Note that this change solves https://issues.jenkins-ci.org/browse/JENKINS-54840, just a friendly reminder to assign/close that issue once this has been released.

**Comment by cloph** [comment]
Splitting the log to a separate file breaks other plugin's features that rely on info being in the actual build log, namely **multijob retry rules** :-(

**Comment by cmur2** [comment]
I am also affected by splitting these logs into a separate file as this behavior is backwards-incompatible and breaks existing configurations with other plugins and not highlighted as such in the release notes.

**Pull request merged**